packages/next/src/server-build.ts

```diff
@@ -1543,6 +1543,39 @@ export async function serverBuild({
       // with that routing section
       ...afterFilesRewrites,
 
+      ...(appDir
```

Comment:
I think we want to keep these above `afterFilesRewrites` so that those can match properly without the extension.